migration.md

```diff
@@ -0,0 +1,267 @@
+## Migration Guide From v4.X to v5.X
+
+### From `Codable struct` to `Codable struct`
+
+Before V4.X, `struct` have to conform to protocol `Codable` to store it as a json string.  
+
+After V5.X, `struct` have to conform to protocol `Defaults.Serializable & Codable` to store it as a json string.  
+
+#### Before migration, your code should be like this
+```swift
+private struct TimeZone: Codable {
+	var id: String
+	var name: String
+}
+
+extension Defaults.Keys {
+	static let timezone = Defaults.Key<TimeZone?>("TimeZone")
+}
+```
+
+#### Migration steps
+1. Let `TimeZone` protocol conform to `Defaults.Serializable`.
+```swift
+private struct TimeZone: Defaults.Serializable, Codable {
+	var id: String
+	var name: String
+}
+```
+1. Now `Defaults[.timezone]` should be readable.
+
+---
+
+### From `Codable enum` to `Codable enum`
+
+Before V4.X, `enum` have to conform to protocol `Codable` to store it as a json string.  
+
+After V5.X, struct have to conform to protocol `Defaults.Serializable & Codable` to store it as a json string.  
+
+#### Before migration, your code should be like this
+```swift
+private enum Period: String, Codable {
+	case tenMinutes = "10 Minutes"
+	case halfHour = "30 Minutes"
+	case oneHour = "1 Hour"
+}
+
+extension Defaults.Keys {
+	static let period = Defaults.Key<Period?>("period")
+}
+```
+#### Migration steps
+1. Let `Period` protocol conform to `Defaults.Serializable`.
+```swift
+private enum Period: String, Defaults.Serializable & Codable {
+	case tenMinutes = "10 Minutes"
+	case halfHour = "30 Minutes"
+	case oneHour = "1 Hour"
+}
+```
+2. Now `Defaults[.period]` should be readable.
+
+---
+
+### From `Codable Array/Dictionary` to `Native Array/Dictionary/Set`(With Native Supported Elements)
+
+Before v4.X, `Defaults` will store array/dictionary as a json string(`["a", "b", "c"]`).  
+
+After v5.X, `Defaults` will store it as a native array/dictionary with native supported elements(`[a, b, c]` ). 
+
+#### Before migration, your code should be like this
+```swift
+extension Defaults.Keys {
+	static let arrayString = Defaults.Key<[String]?>("arrayString")
+	static let dictionaryStringInt = Defaults.Key<[String: Int]?>("dictionaryStringInt")
+	static let dictionaryStringIntInArray = Defaults.Key<[[String: Int]]?>("dictionaryStringIntInArray")
```

Comment:
Can you add an entry using `Set` here too?

Reply:
Do we have tests for deep nesting? For example:

```swift
static let foo = Defaults.Key<[[String: [[String: Int]]]]?>("foo")
```

Reply:
> Can you add an entry using Set here too?

> Do we have tests for deep nesting? For example:

Added!